ngx_http_upstream_check_module.c

```diff
@@ -1153,7 +1165,19 @@ ngx_http_upstream_check_connect_handler(ngx_event_t *event)
     c->write->log = c->log;
     c->pool = peer->pool;
 
-upstream_check_connect_done:
+    if (ucscf->ssl) {
+        ngx_http_upstream_check_ssl_init(peer, ucscf, c);
+        return;
+    }
+
+    ngx_http_upstream_check_connect_done(peer, ucscf, c, rc == NGX_OK);
```

Comment:
It's never `necessary`, but, IMHO, it's better to handle errors in that way. (See http://blog.regehr.org/archives/894.)